iree/turbine/kernel/wave/index_sequence_analysis.py

```diff
@@ -205,14 +206,25 @@ def has_gpr_offsets(node: fx.Node) -> bool:
             for chunk_id in range(num_gpr_chunks):
                 cur_gpr_start_id = chunk_id * gpr_size
                 # Get updated index with VGPR offset.
-                updated_index_with_gpr_offset = {
-                    dim: IndexSequence(
-                        simplified_index[dim].start.subs({GPR_NUM: cur_gpr_start_id}),
-                        gpr_size,
-                        simplified_index[dim].stride,
-                    )
-                    for dim in simplified_index
-                }
+                output_mapping = list(custom.index.keys())
+                if custom.mapping is not None:
+                    output_mapping = list(custom.mapping.output_mapping.keys())
+                # Modify stride to 1 S.T we can have vectorized read/write
+                # iff gpr_offset_dim is or will be (after mapping) fastest dim.
+                updated_gpr_stride = simplified_index[gpr_offset_dim].stride
+                if output_mapping[-1] is gpr_offset_dim:
+                    updated_gpr_stride = 1
+                updated_index_with_gpr_offset = copy(simplified_index)
+                updated_dim_with_gpr_offset = IndexSequence(
+                    updated_index_with_gpr_offset[gpr_offset_dim].start.subs(
+                        {GPR_NUM: cur_gpr_start_id}
+                    ),
+                    gpr_size,
+                    updated_gpr_stride,
+                )
+                updated_index_with_gpr_offset[
+                    gpr_offset_dim
+                ] = updated_dim_with_gpr_offset
```

Comment:
Open to feedback here, but do you think this is more or less readable
```
                updated_index_with_gpr_offset = copy(simplified_index)
                updated_index_with_gpr_offset[gpr_offset_dim] = IndexSequence(
                    updated_index_with_gpr_offset[gpr_offset_dim].start.subs(
                        {GPR_NUM: cur_gpr_start_id}
                    ),
                    gpr_size,
                    (
                        1
                        if output_mapping[-1] is gpr_offset_dim
                        else simplified_index[gpr_offset_dim].stride
                    ),
                )
```

Reply:
Makes sense, done I think (with the `is` becoming `==`)